tensorflow_datasets/scripts/cleanup/delete_old_versions.py

```diff
@@ -0,0 +1,95 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Script to detect all the versions of the various datasets and delete the old/non-existing ones
+
+User is first given previews of all the versions of the datasets with the latest ones in bold
+
+Instructions:
+
+python -m tensorflow_datasets.scripts.delete_old_versions \
+  --data_dir=/path/to/data_dir/
+
+"""
+
+import os
+from absl import flags
+from absl import app
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+
+from tensorflow_datasets.core import constants
+from tensorflow_datasets.core.load import list_full_names
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string("data_dir",constants.DATA_DIR,"Path to the data directory")
+
+def get_redundant_datasets(data_dir):
+  installed_datasets = []
+  all_datasets = tfds.list_builders()
+  exclude = {"downloads","download","manual","extracted"}
+
+  #Get all the non-existing datasets
+  rogue_datasets = [dataset for dataset in tf.io.gfile.listdir(data_dir) if dataset not in all_datasets]
+  rogue_datasets.remove("downloads")
+
+  #Finding all installed datasets
+  for root,dirs,files in os.walk(data_dir, topdown=True):
+    #Excluding the downloads directory and the rogue datasets
+    dirs[:] = [d for d in dirs if d not in exclude]
+    dirs[:] = [d for d in dirs if d not in rogue_datasets]
+    if dirs==[]:
+      installed_datasets.append(root.split("tensorflow_datasets\\")[-1].replace("\\","/"))
```

Comment:
Use os functions for all operations related to file paths

Reply:
Hey @vijayphoenix, using `os.path.split` only gives the tail of the path whereas i would like to get everything after the `~tensorflow_datasets\\`. Could you guide me with it?

Reply:
Try using `os.path.relpath`. (Test the script after the changes)